poseidon2/src/diffusion.rs

```diff
@@ -17,34 +17,9 @@ pub fn matmul_internal<F: Field, const WIDTH: usize>(
     state: &mut [F; WIDTH],
     mat_internal_diag_m_1: [u64; WIDTH],
 ) {
-    match WIDTH {
```

Comment:
I realize this is adapted from the reference impl, but thought it was odd to ignore the diag parameter for 2 & 3. I figure if we want to special case 2 & 3 (not sure there's a real benefit, we could benchmark) we can add a separate struct with a specialized impl of `DiffusionPermutation`.